### PR TITLE
Fix ConstantArrayType for optional/non-optional keys

### PR DESCRIPTION
**Pull request by rvanvelzen**
Fixes phpstan/phpstan#7111

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Perfect, I'm 5 minutes behind you :D Thanks.

File: tests/PHPStan/Analyser/AnalyserIntegrationTest.php [hunk 1]
```diff
@@ -751,6 +751,12 @@ public function testBug7078(): void
 		$this->assertNoErrors($errors);
 	}
 
+	public function testBug3853(): void
+	{
+		$errors = $this->runAnalyse(__DIR__ . '/data/bug-3853.php');
```
Comment (rvanvelzen) [inline tests/PHPStan/Analyser/AnalyserIntegrationTest.php:756]: I reused this test cases which had exactly the same code in it.

**Comment by ondrejmirtes** [comment]
At least I did this: https://github.com/phpstan/phpstan-src/commit/2f49187491aa034d37d2728deb4ee8f5e1102aa6

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**